jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -1108,7 +1078,7 @@ ecma_builtin_typedarray_prototype_sort_compare_helper (ecma_value_t lhs, /**< le
     {
       result = ECMA_NUMBER_MINUS_ONE;
     }
-    else if (lhs_value > rhs_value)
+    else if (lhs_value > rhs_value || (ecma_number_is_zero (rhs_value) && ecma_number_is_negative (rhs_value)))
```

Comment:
What about `rhs <= 0` ? What about nan? `!(rhs < 0)` could hanlde the nan case as well.